src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -252,6 +258,74 @@ export function fetchFacilityDetails(facilityId) {
   };
 }
 
+export function openFacilityPageV2(page, uiSchema, schema) {
+  return async (dispatch, getState) => {
+    try {
+      const initialState = getState();
+      const newAppointment = initialState.newAppointment;
+      const typeOfCareId = getTypeOfCare(newAppointment.data)?.id;
+      const userSiteIds = selectSystemIds(initialState);
+      const useVSP = vaosVSPAppointmentNew(initialState);
+      let parentFacilities = newAppointment.parentFacilities;
+
+      dispatch({
+        type: FORM_PAGE_FACILITY_V2_OPEN,
+      });
+
+      if (!parentFacilities) {
+        parentFacilities = await getOrganizations({
+          siteIds: userSiteIds,
+          useVSP,
+        });
+      }
+
+      const responses = await Promise.all(
+        parentFacilities.map(parent => {
+          const parentId = parent.id;
+          const siteId = parseFakeFHIRId(
+            getIdOfRootOrganization(parentFacilities, parentId),
+          );
+          return getSupportedHealthcareServicesAndLocations({
+            siteId,
+            parentId,
+            typeOfCareId,
+            useVSP,
+          });
+        }),
+      );
+
+      const facilities = responses
```

Comment:
You can concat them without reduce: `[].concat(...row.locations)`.

Also, I know this will change, but I would probably put the sort in the reducer, since it's logic that's not required to fetch the right data.